arc/analysis/funnel_analyzer.py

```
@@ -0,0 +1,473 @@
+            agent_profile: Complete agent profile from normalizer
+            trajectories: List of execution trajectories with results
+            
+        Returns:
+            CapabilityFunnel with detailed step analysis
+        """
+        logger.info(f"Building capability funnel for {len(trajectories)} trajectories")
+        
+        # Extract capabilities from profile
+        capabilities = agent_profile.get("capabilities", {})
+        domains = capabilities.get("domains", ["general"])
+        tool_categories = capabilities.get("tool_categories", {})
```

Ruff (0.11.9)

Local variable `tool_categories` is assigned to but never used

Remove assignment to unused variable `tool_categories`

(F841)